### DISCUSSION TITLE
Supplying default password for unpacking 7zip archives for comparing contents

### DISCUSSION THREAD

**Discussion by fazl** [WinMerge/winmerge#2589] [Q&A] [answered]
Would it be possible to add the ability to give WinMerge default password for it to use when trying to compare a 7zip archive that is password protected, so it doesn't need to ask each time (unless the default password fails, say) ?

WinMerge is showing me a huge number of files with binary differences when comparing two folder trees, containing a lot of files encrypted with 7zip, all using the same password. 

If I manually run a compare on one of these files, and supply the [same] password two times (one for the left, one for the right archive), it then shows me the contents are identical, but the entry in the folder tree is marked as a binary difference.

I couldnt find how I could configure WinMerge (or, its 7zip plugin) with the default password once, then have it use that for each .7z archive automatically when it does a 'binary comparison' all down the folder tree, so that it looks inside archives in the left and right windowpanes to decide they are identical or not..

(This way archive files would be treated logically as folders, that contain other files.)

I have looked at the WinMerge docs and see there are some options for telling it to use the file extension or to look inside the file for a signature to determine the file type, but I've not seen anything about supplying a default password for archives.

I'm using version 2.16.4.4.0 [current version in december 2024].

Well, thanks if you even read this far :)

PS I am aware that there is a switch -mtc which can be added to the 7zip args on creating the archives, to stop 7zip storing some NTFS timestamp (e.g. access time) that can make the archive different for identical content, but that is a separate issue for me.

**Comment by sdottaka** [comment, marked as answer]
Unfortunately, there is currently no such feature in WinMerge.
